sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -232,6 +242,99 @@ export class AssessmentController {
+   * is the first in the sequence to be calculated
+   * for the given student inside the given program year.
+   * If an assessment is identified to be the first in sequence to be calculated
+   * then set calculation start date to hold other assessments until the calculation
+   * is complete.
+   */
+  @ZeebeWorker(Workers.VerifyAssessmentCalculationOrder, {
+    fetchVariable: [ASSESSMENT_ID],
+    maxJobsToActivate: MaxJobsToActivate.Normal,
+  })
+  async verifyAssessmentCalculationOrder(
+    job: Readonly<
+      ZeebeJob<
+        AssessmentDataJobInDTO,
+        ICustomHeaders,
+        VerifyAssessmentCalculationOrderJobOutDTO
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    const jobLogger = new Logger(job.type);
+    try {
+      const assessment = await this.studentAssessmentService.getById(
+        job.variables.assessmentId,
+      );
+      if (!assessment) {
+        const message = "Assessment not found.";
+        jobLogger.error(message);
+        return job.error(ASSESSMENT_NOT_FOUND, message);
+      }
+      const assessmentId = job.variables.assessmentId;
+      const studentId = assessment.application.student.id;
+      const programYearId = assessment.application.programYear.id;
+      jobLogger.log(
+        `Verifying the assessment calculation order for assessment: ${assessmentId} student:${studentId} program year: ${programYearId}.`,
+      );
+      // Check for any assessment with ongoing calculation.
+      const assessmentInCalculationStep =
+        await this.studentAssessmentService.getAssessmentInCalculationStepForStudent(
+          studentId,
+          programYearId,
+        );
+
+      // If an ongoing calculation is happening all other assessments for given student in program year
+      // must wait until the calculation is complete and saved to the system.
+      // Also if a worker job is terminated by any chance after updating the calculation start date
+      // for an assessment, then the assessment must proceed for the calculation.
+      if (
+        assessmentInCalculationStep &&
+        assessmentInCalculationStep.id !== assessmentId
+      ) {
+        jobLogger.log(
+          `There is ongoing calculation happening for assessment: ${assessmentInCalculationStep.id} ` +
+            `and hence assessment : ${assessmentId} is waiting for that calculation to complete.`,
+        );
+        return job.complete({ isReadyForCalculation: false });
+      }
+      jobLogger.log(
+        `There is no ongoing calculation happening while verifying the order for assessment: ${assessment}.`,
+      );
+      // Get the first outstanding assessment waiting for calculation as per the sequence.
+      const [firstOutstandingStudentAssessment] =
+        await this.studentAssessmentService.getOutstandingAssessmentsForStudentInSequence(
+          studentId,
+          programYearId,
+        );
+      jobLogger.log(
```

Comment:
Same from https://github.com/bcgov/SIMS/pull/2745/files#r1470173211

Reply:
Please check the below log as it seems that there is at least one double white space in the middle of the sentence.